src/context.rs

```diff
@@ -111,58 +127,91 @@ impl Context {
 
     pub fn call<
         R: Request,
-        F: for<'a> FnOnce(&'a mut Context, EditorMeta, R::Result) -> () + 'static,
+        F: for<'a> FnOnce(&'a mut Context, EditorMeta, Vec<(ServerName, R::Result)>) -> () + 'static,
     >(
         &mut self,
         meta: EditorMeta,
-        params: R::Params,
+        params: RequestParams<R::Params>,
         callback: F,
     ) where
-        R::Params: IntoParams,
+        R::Params: IntoParams + Clone,
         R::Result: for<'a> Deserialize<'a>,
     {
-        let ops: Vec<R::Params> = vec![params];
+        let ops = match params {
+            RequestParams::All(params) => {
+                let mut ops = Vec::with_capacity(params.len() * self.language_servers.len());
+                for server_name in self.language_servers.keys() {
+                    let params: Vec<_> = params.iter().cloned().collect();
+                    for params in params {
+                        ops.push((server_name.clone(), params));
+                    }
+                }
+                ops
+            }
+            RequestParams::Each(params) => params
+                .into_iter()
+                .flat_map(|(key, ops)| {
+                    let ops: Vec<(ServerName, <R as Request>::Params)> =
+                        ops.into_iter().map(|op| (key.clone(), op)).collect();
+                    ops
+                })
+                .collect(),
+        };
         self.batch_call::<R, _>(
             meta,
             ops,
             Box::new(
-                move |ctx: &mut Context, meta: EditorMeta, mut results: Vec<R::Result>| {
-                    if let Some(result) = results.pop() {
```

Comment:
The old results.pop() seems super weird. Why would we not consume all of results?
Maybe it's just an `Option<R::Result>` in disguise...
Anyway, the new version seems fine though it would be good to understand this.
Maybe it's because we have `batch_sizes()` now.